src/similarity.ts

```diff
@@ -120,19 +206,19 @@ function countStringDifferences(
   return numberOfDifferentCharacters;
 }
 
-function pathWithoutQueryParameters(path: string) {
+function pathWithoutQueryParameters(path: string): string {
   const questionMarkPosition = path.indexOf("?");
   if (questionMarkPosition !== -1) {
-    return path.substr(0, questionMarkPosition);
+    return path.substring(0, questionMarkPosition);
   } else {
     return path;
   }
 }
 
-function queryParameters(path: string) {
+function parseQueryParameters(path: string): ParsedUrlQuery {
   const questionMarkPosition = path.indexOf("?");
   if (questionMarkPosition !== -1) {
-    return queryString.parse(path.substr(questionMarkPosition));
+    return parseQueryString(path.substring(questionMarkPosition + 1));
```

Comment:
The ` + 1` addition here fixes a long standing bug that I found while swapping the query string parsing functionality from the long deprecated library to the builtin NodeJS library. Before, the substring returned included in the `?` at the start of the string as the slice was starting at the index of the `?`. Apparently the old library accepted this without issue, but the new NodeJS one did not.